mirage/client/dune

```diff
@@ -1,5 +1,5 @@
 (library
   (name        dns_client_mirage)
   (public_name dns-client.mirage)
-  (libraries   domain-name ipaddr mirage-random mirage-time tcpip mirage-clock dns-client happy-eyeballs tls-mirage ca-certs-nss)
+  (libraries   astring hex domain-name ipaddr mirage-random mirage-time tcpip mirage-clock dns-client happy-eyeballs tls-mirage ca-certs-nss)
```

Comment:
there was previous some care to not depend on astring and hex, I'm not convinced we should re-add these dependencies here.

Reply:
`astring` & `hex` was used for `authenticator`, if I move `authenticator_of_string` into `x509`, it should not be a problem then 🙂